### PR TITLE
docs(data-model): file headers for the `test` tree's top level

### PR DESCRIPTION
**Pull request by danfuzz**
I (agent working on behalf of @danfuzz) am finishing the file-header
conformance work in `packages/data-model`, against the "File headers" section
of `docs/development/code-comment-style.md`. This covers the `test` tree's top
level. Its subdirectories are #5702 and `src` is #5700; with this one the
package is done.

Nineteen headers written from scratch, plus `value-debug-indented.test.ts`
converting its `//` block to a doc comment with its prose crossing unchanged.

### What the headers say

Each states the property its cases are arranged around:

- `data-uri-codec.test.ts` — a minted URI has to be canonical and still
  discriminating, which is where `-0` against `+0`, the two infinities, and
  every `NaN` minting a single URI earn their place.
- `cell-rep.test.ts` — each regime has to *refuse* the other's form, not merely
  produce its own. That refusal is what stops a value written under one setting
  from being quietly misread under the other.
- `deep-freeze.test.ts` — an answer is cached by identity, so a wrong `true` is
  permanent rather than momentary, which is why several cases exist only to pin
  when a result may be cached.
- `native-conversion.test.ts` — refusal is central rather than incidental:
  conversion is a vetting boundary, and a value that cannot be represented is
  rejected rather than approximated into something that merely looks
  representable.
- `cloneIfNecessary.test.ts` — two option combinations are errors rather than
  choices, having no coherent answer to give.

### One header records a gap rather than a guarantee

`value-hash.test.ts` has a group asserting the behavior of types that are not
handled yet. The header says so, so the gap sits visibly in the suite rather
than being merely absent from it.

### Two claims walked back before they shipped

Both were quantitative, and both would have been the kind of sentence that
reads fine and is false:

- The scalar groups in `value-hash.test.ts` are *not* the largest — its
  edge-case group spans more lines. The size claim is gone.
- Refusals are not "most" of `native-conversion.test.ts`; they are roughly
  eight of its forty-odd groups. Rewritten as a claim about the design rather
  than a count.

The one count that survived was checked: the `toJSON()` group in
`native-conversion.test.ts` really does appear three times, once per entry
point, verified at all three positions rather than inferred from two.

### Testing

`deno task test` in `packages/data-model`: `ok | 52 passed (2340 steps) |
0 failed`. Note that `deno task check` does not cover this package, so that
suite is the gate. `deno fmt --check` and `deno lint` are clean over the twenty
files, no added line exceeds 80 characters measured in characters rather than
bytes, and the one form conversion was checked against its committed version by
comparing the two prose word-bags.

<a href="https://cubic.dev/pr/commontoolsinc/labs/pull/5703?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/data-model/test/valueEquals.test.ts [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+ * Two values are comparable only once they are the same kind of thing, so a
+ * plain object and an array are unequal without their contents being consulted
+ * at all, as are two fabric values of different concrete classes. The cases
+ * walk that branch deliberately rather than sampling it.
+ *
+ * Frozen state must not change an answer, which is what the matrix over it is
+ * for: equality is about what a value holds, not about whether it can still be
+ * written to.
+ *
+ * Signed zeros and the non-finite numbers get their own group, being exactly
```
[the remark this answers is not in the document]
Comment (danfuzz) [reply packages/data-model/test/valueEquals.test.ts]: Confirmed and fixed in `6fec4050c`. I (agent working on behalf of @danfuzz) checked the group's assertions rather than reasoning about it, and you are right on the specifics:

- `valueEqual(-0, 0)` is `false` where `===` is `true` — diverges.
- `valueEqual(NaN, NaN)` is `true` where `===` is `false` — diverges.
- The three infinity assertions (`Infinity`/`Infinity`, `-Infinity`/`-Infinity`, `Infinity`/`-Infinity`) all match what `===` gives — no divergence.

So "exactly" was false for a third of the group. Rather than only dropping the qualifier, the header now says what the infinities are actually doing there: they are the counterweight, pinning that the comparison has not over-corrected while fixing the two cases that genuinely diverge. That seemed more useful than narrowing the sentence to omit them.
